pkg/operator/controller/canary/controller.go

```diff
@@ -117,9 +167,122 @@ type reconciler struct {
 	client client.Client
 }
 
-// TODO: Canary Controller Phase 2
-// Add callers for these 2 functions
-//
+func (r *reconciler) startCanaryRoutePolling(stop <-chan struct{}) error {
+	// Keep track of how many canary checks have passed
+	// so the route endpoint can be periodically cycled.
+	checkCount := 0
+
+	// Keep track of successive canary check failures
+	// for status reporting.
+	successiveFail := 0
+
+	// Signal that the canary polling loop is running.
+	routeProbeRunning = true
+
+	go wait.Until(func() {
+		// Get the current canary route every iteration in case it has been modified
+		haveRoute, route, err := r.currentCanaryRoute()
+		if err != nil {
+			log.Error(err, "failed to get current canary route for canary check")
+			return
+		} else if !haveRoute {
+			log.Info("canary check route does not exist")
+			return
+		}
+		// Periodically rotate route endpoint after every 5 successful checks.
+		if checkCount == 6 {
+			haveService, service, err := r.currentCanaryService()
+			if err != nil {
+				log.Error(err, "failed to get canary service")
+				return
+			} else if !haveService {
+				log.Info("canary check service does not exist")
+				return
+			}
+			route, err = r.rotateRouteEndpoint(service, route)
+			if err != nil {
+				log.Error(err, "failed to rotate canary route endpoint")
+				return
+			}
+			checkCount = 0
+			// Give the router time to reload by returning here.
+			return
+		}
+
+		success, err := probeRouteEndpoint(route)
+		host := route.Spec.Host
+		if !success {
+			log.Error(err, "error performing canary route check")
+			SetCanaryRouteReachableMetric(host, false)
+			successiveFail += 1
+			// Mark the default ingress controller degraded after 5 successive canary check failures
+			if successiveFail > 5 {
```

Comment:
As mentioned above, consider using a variable (maxFailures?)  to store the number of failures to compare to successiveFails.

Reply:
Fixed. Thanks!